WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -501,12 +501,20 @@ private BobClient CreateBobClient(RoundState roundState)
 
 	private bool SanityCheck(IEnumerable<TxOut> expectedOutputs, Transaction unsignedCoinJoinTransaction)
 	{
-		var coinJoinOutputs = unsignedCoinJoinTransaction.Outputs.Select(o => (o.Value, o.ScriptPubKey));
-		IEnumerable<(Money TotalAmount, Script Key)>? expectedOutputTuples = expectedOutputs
-			.GroupBy(x => x.ScriptPubKey)
-			.Select(o => (o.Select(x => x.Value).Sum(), o.Key));
+		var coinJoinOutputs = unsignedCoinJoinTransaction.Outputs;
+		var coinjoinOutputScripts = coinJoinOutputs.Select(x => x.ScriptPubKey);
+		var expectedOutputScripts = expectedOutputs.Select(x => x.ScriptPubKey);
+		var allMyScriptsAreThere = coinjoinOutputScripts.IsSuperSetOf(expectedOutputScripts);
 
-		return coinJoinOutputs.IsSuperSetOf(expectedOutputTuples);
+		var expectedScriptHashSet = coinjoinOutputScripts.ToHashSet();
```

Comment:
```suggestion
		var expectedScriptHashSet = expectedOutputScripts.ToHashSet();
```

Reply:
Yes, this was a bug but the code was wrong in other way because allowed manipulation of the outputs. I changed that and add tests.